src/components/PieChart.js

```diff
@@ -0,0 +1,161 @@
+
+import React from 'react';
+import Highcharts from 'highcharts';
+import drilldown from 'highcharts/modules/drilldown.js';
+import noDataToDisplay from 'highcharts/modules/no-data-to-display.js';
+
+function getCategoriesSeriesData(datas) {
+	datas = datas[0].data;
+	datas.sort(function(a, b){return b[1] - a[1]});
+
+	if (datas.length === 0) {
+		return []
+	}
+
+	const series = [];
+	const cutoffIndex = getCutoffIndex(datas);
+	let count = 0;
+
+	for (let i = 0; i < cutoffIndex; i += 1) {
+		series.push({
+			name: datas[i][0],
+			y: datas[i][1]
+		});
+	}
+
+	for (let i = cutoffIndex; i < datas.length; i += 1) {
+		count += datas[i][1];
+	}
+
+	series.push({
+		name: 'Other',
+		y: count,
+		drilldown: 'Other'
+	});
+	return series;
+}
+
+function getCategoriesDrilldownData(datas) {
+	datas = datas[0].data;
+	datas.sort(function(a, b){return b[1] - a[1]});
+
+	if (datas.length === 0) {
+		return []
+	}
+
+	const series = [];
+	const cutoffIndex = getCutoffIndex(datas);
+	const cutoff = parseInt(datas[cutoffIndex][1] * 0.4, 10)
+	let count = 0;
+
+	for (let i = cutoffIndex; i < datas.length; i += 1) {
+		if (datas[i][1] <= cutoff) {
+			count += datas[i][1];
+		} else {
+			series.push([datas[i][0],datas[i][1]]);
+		}
+	}
+
+	series.push(['Other', count]);
+	return series;
+}
+
+function getCutoffIndex(datas) {
+	const total = datas.reduce((total, arr) => total + arr[1], 0);
```

Comment:
just comments here again. It's hard to jump into this code and follow it since I don't know what arr[1] is.